framework/PostProcessors/TopologicalDecomposition.py

```diff
@@ -116,47 +117,27 @@ def inputToInternal(self, currentInp):
-      currentInput = currentInp [-1]
-    else:
-      currentInput = currentInp
-    if type(currentInput) == dict:
-      if 'features' in currentInput.keys():
-        return currentInput
-    inputDict = {'features':{}, 'targets':{}, 'metadata':{}}
-    if hasattr(currentInput, 'type'):
-      inType = currentInput.type
-    elif type(currentInput).__name__ == 'list':
-      inType = 'list'
-    else:
+    # TODO typechecking against what currentInp can be; so far it's a length=1 list with a dataobject inside
+    currentInp = currentInp[0]
+    currentInp.asDataset()
+    # nowadays, our only input should be DataObject
+    ## if no "type", then you're not a PointSet or HistorySet
+    if not hasattr(currentInp,'type') or currentInp.type != 'PointSet':
       self.raiseAnError(IOError, self.__class__.__name__,
-                        ' postprocessor accepts files, HDF5, Data(s) only. ',
-                        ' Requested: ', type(currentInput))
-
-    if inType not in ['HDF5', 'PointSet', 'list'] and not isinstance(currentInput,Files.File):
-      self.raiseAnError(IOError, self, self.__class__.__name__ + ' post-processor only accepts files, HDF5, or DataObjects! Got ' + str(inType) + '!!!!')
-    # FIXME: implement this feature
-    if isinstance(currentInput,Files.File):
-      if currentInput.subtype == 'csv':
-        pass
-    # FIXME: implement this feature
-    if inType == 'HDF5':
-      pass  # to be implemented
-    if inType in ['PointSet']:
-      for targetP in self.parameters['features']:
-        if   targetP in currentInput.getParaKeys('input'):
-          inputDict['features'][targetP] = currentInput.getParam('input' , targetP)
-        elif targetP in currentInput.getParaKeys('output'):
-          inputDict['features'][targetP] = currentInput.getParam('output', targetP)
-      for targetP in self.parameters['targets']:
-        if   targetP in currentInput.getParaKeys('input'):
-          inputDict['targets'][targetP] = currentInput.getParam('input' , targetP)
-        elif targetP in currentInput.getParaKeys('output'):
-          inputDict['targets'][targetP] = currentInput.getParam('output', targetP)
-      inputDict['metadata'] = currentInput.getAllMetadata()
-    # now we check if the sampler that genereted the samples are from adaptive... in case... create the grid
-    if 'SamplerType' in inputDict['metadata'].keys():
-      pass
+                        ' postprocessor only accepts PointSet DataObjects for input. ',
+                        ' Requested: ', type(currentInp))
+    # now we know we have a PointSet
+    ## TODO FIXME maintaining old structure for now, in the future convert to use DataObject directly
+    ##    and not bother with inputToInternal
+    ##    This works particularly well since we only accept point sets.
+    data = currentInp.asDataset(outType='dict')['data']
+    inputDict = {'features':dict((var,np.array(data[var],dtype=float)) for var in self.parameters['features']),
```

Comment:
you should not need the casting here anymore right?

Reply:
True!  Fixed.